source/simulator/solver.cc

```diff
@@ -747,7 +747,7 @@ namespace aspect
         SolverFGMRES<LinearAlgebra::BlockVector>
         solver(solver_control_cheap, mem,
                SolverFGMRES<LinearAlgebra::BlockVector>::
-               AdditionalData(30, true));
+               AdditionalData(50, true));
```

Comment:
I don't recall the details but does a restart length of 50 mean that the GMRES cycles over *in* the 50th iteration, or *after* the 50th iteration? If it was the former, it may be worth setting the restart length to 51 to avoid cycling just before the end of the 200th iteration that we run. (That is of course not a new issue -- it's the same with restart=30 if we run 30 iterations. I've long wondered if that is the correct number, or if we should set restart=31.)